file-formats/dsfi/type/zif_aff_dsfi_v1.intf.abap

```diff
@@ -0,0 +1,43 @@
+INTERFACE zif_aff_dsfi_v1
+  PUBLIC.
+
+
+  "! <p class="shorttext">Engine</p>
+  "! Engine
+  "! $values {@link if_aff_dsfi_v1.data:co_engine}
+  TYPES: ty_engine TYPE c LENGTH 3.
+
+  CONSTANTS:
+    BEGIN OF co_engine,
+      "! <p class="shorttext">Analytical Engine</p>
+      "! Analytical engine
+      analytical_engine TYPE ty_engine VALUE 'ANA',
+    END OF co_engine.
+
+
+  TYPES:
+    BEGIN OF ty_dsfi,
+      "! <p class="shorttext">Scalar Function</p>
+      "! Scalar function
+      scalar_function_name TYPE c LENGTH 30,
+      "! <p class="shorttext">Engine</p>
+      "! Engine
+      engine               TYPE ty_engine,
+    END OF ty_dsfi.
+
+  TYPES:
+    "! <p class="shorttext">ABAP file formats for object type DSFI</p>
+    "! DSFI object type
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60_src,
+      "! <p class="shorttext">General Information</p>
+      "! $required
+      content        TYPE ty_dsfi,
```

Comment:
You can put the fields scalar_function_name and engine directly into ty_main. You dont have to put them into the intermediate structure ty_dsfi. 

If you want to have the title "General Information" for the two fields in the editor you have to remove the title of ty_main in line 29
""! <p class="shorttext">ABAP file formats for object type DSFI</p>"
then automatically "General Information" will be used

Reply:
@schneidermic0 
what is your opinion about the field "content"?  Should the  fields "name" and "engine" be directly in ty_main?

Reply:
From my point of view specific sub sections are fine. We tried to avoid generic sections like `content` in other ABAP file formats. Therefore, I would move the fields either directly to `ty_main` or rename `content` to a more specific name.

Reply:
The idea and motivation is to have a single type, that contains the information of the DSFI object.
So the ABAP dev that uses the type writes
`if_aff_dsfi_v1=>ty_main-content` and then, knows that he/she is in the DSFI specific part. Hence, does not get distracted by `if_aff_dsfi_v1=>ty_main-header` or other general parts.

I will reach out to the object owner and maybe we find a better name than `content`.

Reply:
Decision: no nesting needed here. Removed the `content` in [6fb06fa](https://github.com/SAP/abap-file-formats/pull/319/commits/6fb06fac3fecb6d2954199771fccd23adcb5a7c9)